packages/core/src/bar-chart.ts

```diff
@@ -314,7 +315,11 @@ export class BarChart extends BaseAxisChart {
 				const tooltipRef = select(self.holder).select("div.chart-tooltip");
 
 				const relativeMousePosition = mouse(self.holder as HTMLElement);
-				tooltipRef.style("left", relativeMousePosition[0] + Configuration.tooltip.magicLeft2 + "px")
+				tooltipRef.style(
+					"left",
+					!rtl ?
+						relativeMousePosition[0] - Configuration.tooltip.magicLeft2 + "px" :
+							relativeMousePosition[0] - (tooltipRef.node() as Element).clientWidth - Configuration.tooltip.magicLeft2 + "px")
```

Comment:
Again, too complex to read, try storing things in variables more

Reply:
For bar-chart in general, do we still need setXScale(), more so why are bar charts overriding base-axis charts?